src/legacy/ui/public/vislib/visualizations/point_series/column_chart.js

```diff
@@ -220,10 +269,29 @@ export class ColumnChart extends PointSeries {
       if ((isHorizontal && d.y < 0) || (!isHorizontal && d.y > 0)) {
         return yScale(0);
       }
-
       return yScale(d.y);
     }
 
+    function labelX(d, i) {
+      return x(d, i) + widthFunc(d, i) / 2;
+    }
+
+    function labelY(d) {
+      if (isHorizontal) {
+        return d.y >= 0 ? y(d) - 4 : y(d) + heightFunc(d) + this.getBBox().height;
+      }
+      return d.y >= 0 ? y(d) + heightFunc(d) + 4 : y(d) - this.getBBox().width - 4;
+    }
+
+    function labelDisplay(d, i) {
+      if (isHorizontal && this.getBBox().width > widthFunc(d, i)) {
+        return 'none';
+      }
+      if (!isHorizontal && this.getBBox().height > widthFunc(d)) {
+        return 'none';
+      }
+      return 'block';
+    }
```

Comment:
Specially with small bars or with bigger values, there is the issue that only lower bars or lower values are displayed. like in this screenshot
<img width="804" alt="Screenshot 2019-06-18 at 12 14 36" src="https://user-images.githubusercontent.com/1421091/59674562-d33ab100-91c3-11e9-9f98-d92a1d5a8e48.png">
What do you think if we can add a configurable parameter that force the display of all the labels, also if they are bigger than the bar width?
As for the stacked case, just be aware of the color of the text for non stacked bars because the text can overlap with a bar with the same color.
Maybe we can just use the black/white color also for non stacked bars

Reply:
I'm afraid that if we allow to label to spread out, it may become a mess.  Even, with a configuration (which is a user responsiblity), still at configuration time, it may remained nice, but later with specific set of data, the graph become ugly.